DietPi-WiFiDB | Enhancements

MichaIng opened the pull request:
**Status**: Testing
🈯️ Native x86_64 Stretch
- [ ] A quick test on Jessie would be great with double quote inside SSID and/or key. Just to assure that .conf parsing did not change in between.

**Reference**: https://github.com/Fourdee/DietPi/pull/2177#issuecomment-433071910

**Commit list/description**:
+ DietPi-WiFiDB | Switch to store WiFiDB with single quotes to reduce escape effort
  - Tested it up and down on notebook with Stretch. Script internal input box and variable handling does not need any escaping. Storing strings to source-able `dietpi-wifi.db` requires "escaping" of single quote only, which is only possible via `'\''` to close single quoted literal string, insert escaped single quote and reopen literal single quoted string till the end. Inside `wpa_supplicant.conf`, double quotes for SSID, key, user and password are required, but no escaping, not even of double quote, required, since the values are parsed from first till last double quote 😃, intelligent solution. Single and double quotes and backslashes within bash variable manipulation `${VAR//<old>/<new}` need to be escaped by backslash. This is then how the code came out, e.g.: `'${aWIFI_SSID[$i]//\'/\'\\\'\'}'`
+ DietPi-WiFiDB | Do required escaping script internally, to save user from doing this and us to maintain documentation about it
+ DietPi-WiFiDB | Only ask for WiFi key, if SSID and key manager was successfully chosen, otherwise reset values of the slot and return to slot overview
  - On the long term we could also implement a completed 'manual' + slot menu where one can select SSID, key manager, key and as well `EAP` settings, in case chosen as key manager. 
+ DietPi-WiFiDB | Set ERROR_CODE=0 as default, as currently it always exits with ERROR_CODE=1, if valid INPUT was set.
  - As said, currently exit code is always `1`, only `0` if invalid input was given. I guess this should have been the other way round 😆. But at least should not break anything, since this error code is never used so far.
+ DietPi-WiFiDB | Minor wording and coding

MichaIng committed:
v6.18

+ DietPi-WiFiDB | Switch to store WiFiDB with single quotes to reduce escape effort
+ DietPi-WiFiDB | Do required escaping script internally, to save user from doing this and us to maintain documentation about it
+ DietPi-WiFiDB | Only ask for WiFi key, if SSID and key manager was successfully chosen, otherwise reset values of the slot and return to slot overview
+ DietPi-WiFiDB | Set ERROR_CODE=0 as default, as currently it always exits with ERROR_CODE=1, if valid INPUT was set.
+ DietPi-WiFiDB | Minor wording and coding

MichaIng commented:
Note: `dietpi-wifi.db` does not need to be adjusted during patch. Since double quoted values are already escaped correctly, sourcing the db leads to correct values and will be automatically written back with single quotes including correct escaping.

Fourdee commented on dietpi/func/dietpi-wifidb:
```diff
@@ -259,24 +253,22 @@ _EOF_
 		G_WHIP_DEFAULT_ITEM="${aWIFI_KEYMGR[$WIFI_SSID_INDEX]}"
-		G_WHIP_MENU 'Please select a WiFi encryption mode.\n\nNB: If unsure, its most likely WPA-PSK.'
-		if (( $? == 0 )); then
+		G_WHIP_MENU 'Please select a WiFi encryption mode.\n\nNB: If unsure, its most likely WPA-PSK.'; local EXIT_CODE=$?
```
if local should be lowercase.

Init following at start of function?
```
local return_code=0
```

Then use ```return_code``` here? and ```$return return_code```?

MichaIng replied:
@Fourdee 
Ah yes fits better, non-capital letters since local and this no real exit code.
€: Done

MichaIng committed:
v6.18

+ DietPi-WiFiDB | Non-capital, renamed and pre-declared return code: https://github.com/Fourdee/DietPi/pull/2193#discussion_r229350445

Fourdee commented:
@MichaIng 

Great work, looks good 👍 Lets merge for testing.
